### DISCUSSION TITLE
Sampling neighboring edges by using weights

### DISCUSSION THREAD

**Discussion by pbelmann** [pyg-team/pytorch_geometric#7662] [Q&A]
Hi,

is there a way to do weighted sampling of edges using pyg? I'm searching for a function similar to DGLs `dgl.sampling.sample_neighbors` method (See `prob` parameter).
https://docs.dgl.ai/en/0.8.x/generated/dgl.sampling.sample_neighbors.html

**Comment by wsad1** [comment]
This isn't supported at the moment. But there is work underway to support that in `NeighborLoader`, potentially in the next release. 
https://github.com/pyg-team/pytorch_geometric/issues/5345.

**Reply by rusty1s** [reply]
Yes, here is the broader roadmap: https://github.com/pyg-team/pytorch_geometric/issues/7331
